onion-messaging.md

```diff
@@ -0,0 +1,143 @@
+## MESSAGE FORMAT USED on the ONION MESSAGING CHANNEL
+
+( `||` means concatenation for strings, here)
+
+These message channels are based on communication between Joinmarket nodes, where two classes of node, namely *directory nodes* and *maker* nodes, are accessible via Tor v3 onion addresses, and the remaining nodes connect to them (using a SOCKS5 proxy).
+
+Messages are defined as in `jmdaemon.onionmc.OnionCustomMessage`. They are an encoded json struct, which always contains two fields: `line` and `type`. `line` is always a `str` and `type` is always an `int`. This `type` is the integer specified in this file as `*_MESSAGE_TYPE`s.
```

Comment:
Question: why `line` and not `msg`, like in the code, or something else like that? Asking because `line` can be confused with the `lines` from `LineReceiver`.

Reply:
Tend to agree but it doesn't seem like too big of a deal. I'll leave it for now.